smithy-swift-codegen/src/test/kotlin/HttpBodyMiddlewareTests.kt

```diff
@@ -155,9 +155,17 @@ class HttpBodyMiddlewareTests {
                             let payload1body = ClientRuntime.HttpBody.data(payload1data)
                             input.builder.withBody(payload1body)
                         } else {
-                            let payload1data = try encoder.encode(input.operationInput)
-                            let payload1body = ClientRuntime.HttpBody.data(payload1data)
-                            input.builder.withBody(payload1body)
```

Comment:
What was happening before?  Was an empty body failing to encode?  Or was there actually some use in encoding `operationInput`?

Reply:
Previously, it would attempt to encode the `operationInput`. The smithy specification does NOT dictate that the input struct be encoded if the http payload member is nil. Therefore, this can produce unexpected behavior.

For JSON, this would be fine (which i assume is why this code existed in the first place), as it would encode as `{}`.
But for XML it would produce an unexpected result, since it would encode the input struct such as `<SomeInput></SomeInput>` This is the problem that the issue filed describes https://github.com/awslabs/aws-sdk-swift/issues/580